typescript/infra/scripts/warp-routes/monitor-warp-routes-balances.ts

```diff
@@ -265,12 +266,37 @@ async function checkBalance(
               return parseFloat(
                 ethers.utils.formatUnits(syntheticBalance, token.decimals),
               );
+            }
             case ProtocolType.Cosmos:
               // TODO - cosmos synthetic
               return 0;
           }
           break;
         }
+        case TokenType.XERC20: {
+          switch (token.protocolType) {
+            case ProtocolType.Ethereum: {
+              if (!token.tokenAddress)
+                throw new Error('Token address missing for xERC20 token');
+              const provider = multiProtocolProvider.getEthersV5Provider(chain);
+              const hypXERC20 = HypXERC20__factory.connect(
+                token.hypAddress,
+                provider,
+              );
+              const xerc20Address = await hypXERC20.wrappedToken();
+              const xerc20 = IXERC20__factory.connect(xerc20Address, provider);
+              const syntheticBalance = await xerc20.totalSupply();
```

Comment:
@yorhodes @tkporter - I think this would assume that we are the only bridge that can mint on this xerc20, I cant see a way where we can the read the balance of tokens that we have minted

Reply:
mmmm this is interesting, I also don't know of a way to get the tokens we have been responsible for minting.

Is this an issue with fastUSD or ezETH?

Reply:
So the synthetic token on SEI is an xERC20 so it impacts both

Reply:
are there other minters/burners on them in practice or is it just us atm?

Reply:
It is just us atm